### PR TITLE
feat: add proposal simulations for MsgRecoverClient and MsgIBCSoftwareUpgrade

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #4599 

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/simulation/proposals.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	params := types.DefaultParams()
+	params.AllowedClients = []string{"06-solomachine", "07-tendermint"}
+
+	return &types.MsgUpdateParams{
+		Signer: signer.String(),
+		Params: params,
+	}
+}
+
+// SimulateClientMsgRecoverClient returns a random MsgRecoverClient for 02-client
```
Comment (colin-axner) [inline modules/core/simulation/proposals.go:67]: haha not so random, but I can dig it

File: modules/core/simulation/proposals.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	var signer sdk.AccAddress = address.Module("gov")
+
+	chainID := "chain-a-0"
+	trustingPeriod := time.Hour * 24 * 7 * 2
+	// two week unbonding period instead of default three week unbonding period
+	ubdPeriod := time.Hour * 24 * 7 * 2
+	maxClockDrift := time.Second * 10
+	upgradePath := []string{"upgrade", "upgradedIBCState"}
+
+	upgradedClientState := ibctm.NewClientState(chainID, ibctm.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, types.ZeroHeight(), commitmenttypes.GetSDKSpecs(), upgradePath)
```
Comment (colin-axner) [inline modules/core/simulation/proposals.go:89]: technically we only need to set the chain specific fields:
- chainID
- unbonding period
- proof specs
- latest height
- upgrade path

Maybe we remove the trusting period and the max clock drift?

**Review by colin-axner — APPROVED** [review]
LGTM! I reviewed via this [comparison](https://github.com/cosmos/ibc-go/compare/feat/govv1...charly/issue%234599-add-proposal-simulations-for-msgrecoverclient-and-msgibcsoftwareupgrade)

File: modules/core/simulation/proposals.go (hunk 1 above)
Comment (charleenfei) [reply modules/core/simulation/proposals.go:67]: 😂

File: modules/core/simulation/proposals.go (hunk 2 above)
Comment (charleenfei) [reply modules/core/simulation/proposals.go:89]: you mean, from the `ibctm.NewClientState` method?

**Review by damiannolan — APPROVED** [review]

**Review by colin-axner — APPROVED** [review]
ACK

**Pull request merged**